src/exml.erl

```diff
@@ -53,6 +57,34 @@ xml_size({Key, Value}) ->
     + 4 % ="" and whitespace before
     + byte_size(Value).
 
+%% @doc Sort a (list of) `xmlel()`.
+%%
+%% Sorting is defined as calling `lists:sort/1` at:
+%% * all the `xmlel's provided (if there is a list of them) AND
+%% * all the `xmlel' elements' attributes recursively (the root and descendants) AND
+%% * all the `xmlel' children recursively (the root and descendants).
+%% The order is ascending.
+%%
+%% The implementation of this function is subtle modification of
+%% https://github.com/erszcz/rxml/commit/e8483408663f0bc2af7896e786c1cdea2e86e43d
+-spec xml_sort(item() | [item()]) -> item() | [item()].
+xml_sort(#xmlcdata{} = Cdata) ->
+    Cdata;
+xml_sort(#xmlel{} = El) ->
+    #xmlel{ attrs = Attrs, children = Children } = El,
+    El#xmlel{
+      attrs = lists:sort(Attrs),
+      children = lists:sort([ xml_sort(C) || C <- Children ])
+     };
+xml_sort(#xmlstreamstart{ attrs = Attrs } = StreamStart) ->
+    StreamStart#xmlstreamstart{ attrs = lists:sort(Attrs) };
+xml_sort(#xmlstreamend{} = StreamEnd) ->
+    StreamEnd;
+xml_sort(Elements) when is_list(Elements) ->
+    lists:sort([ xml_sort(E) || E <- Elements ]);
+xml_sort(Unknown) ->
+    Unknown.
```

Comment:
I'm wondering if we need that at all. According to the function spec everything is covered and we never get to this clause (as long as valid item() is passed).

Reply:
I remember adding it for the sake of convenience, when I was figuring out how to return stuff from the Rust parser to the Erlang layer and I returned some rubbish just to test the mechanism. It might not be needed outside experimental code and even if it would, it's trivial to add again.